sanic/app.py

```diff
@@ -700,13 +700,17 @@ def url_for(self, view_name: str, **kwargs):
         query_string = urlencode(kwargs, doseq=True) if kwargs else ""
 
         if external and port:
-            port = int(port)
-            if (
-                (scheme == "http" and port != 80)
-                or (scheme == "https" and port != 443)
-                or (scheme in ("ftp", "ftps") and port != 21)
-            ):
-                netloc = "{}:{}".format(netloc, port)
+            try:
+                port = int(port)
+            except (ValueError, TypeError):
+                pass  # just ignore it
```

Comment:
Yeah, I think we can just ignore this (for now). If someone complains in the future, we can perhaps just add a `logger.warn` call here and that should suffice.

Reply:
Don't know why lint check failed.

Reply:
@lixxu try to run `black` to format the code (in the root of the repo):
```
$ black ./sanic
```

Reply:
oops, 69 files reformatted, 19 files left unchanged. Should I commit these changes? a lot files, may conflict with other PRs.

Reply:
Have you rebased with the last master?

Reply:
sorry, I ran the cmd outside of sanic folder, now only the 2 files updated. :)

Reply:
Ohh, that makes sense :smile:

Reply:
TLDR; Fail fast and fail loud.
I saw that @lixxu had already implemented it first without try:except then they were asked to put it in a try:except block. In this case, I believe failing loudly would have better point that failing silently here. A user might wrongly pass some object that can't be converted into integer and this function will fail silently.